tests/integration/temporal_client/workflows.py

```diff
@@ -28,3 +30,48 @@ async def run(self, name: str) -> str:
             Workflow execution
         """
         return await workflow.execute_activity(say_hello, name, schedule_to_close_timeout=timedelta(seconds=5))
+
+
+@workflow.defn
+class GreetingWorkflow:
+    """Temporal workflow class."""
+
+    def __init__(self) -> None:
+        """Construct."""
+        self._pending_greetings: asyncio.Queue[str] = asyncio.Queue()
+        self._exit = False
+
+    @workflow.run
+    async def run(self) -> List[str]:
+        """Workflow execution method.
+
+        Returns:
+            Workflow execution result.
+        """
+        # Continually handle from queue or wait for exit to be received
+        greetings: List[str] = []
+        while True:
+            # Wait for queue item or exit
+            await workflow.wait_condition(lambda: not self._pending_greetings.empty() or self._exit)
```

Comment:
Out of curosity, how does it wait under-the-hood for the variable change? If you're aware.

Reply:
This is the implementation I found in their [SDK](https://github.com/temporalio/sdk-python/blob/a4224a20189c8c8078edbc66482c31860193c1b3/temporalio/workflow.py#L709), hope it helps!